docs/features/role-based-access-control/high-level-design.md

```diff
@@ -0,0 +1,76 @@
+# Role base access control - High-level design
+
+This document aims at providing an overview of the architecture of Role-based Access Control (RBAC from here on) for
+the Marxan platform.
+
+## Roles
+
+**Platform admins** should be identified in a database table dedicated for the purpose. Adding users to the admins
+DB table should only be possible for other admin users.
+
+The remaining roles supported by the platform (`Owner`, `Contributor` and `Viewer`) will be stored in the `roles` 
+database in the `marxan-api` database. These roles should cover the roles defined in the project requirements:
+
+* "Platform admin" would be achieved with the implementation of a dedicated DB table for platform admins;
+* "Project owner" -> role Owner for resource Project;
+* "Project contributor" -> role Contributor for resource Project;
+* "Project reviewer (read-only)" -> role Viewer for resource Project;
+* "Project solution-viewer" -> role Viewer for resource "Solutions";
+
+There is an implicit relationship between platform actions and the supported roles. By default, it is assumed that:
+* a user with role `Viewer` can read the information for the given resource (which usually includes calls to 
+`GET resource` and `GET resource/:id` endpoints).
+* user with role `Contributor` can perform all the actions a `Viewer` can, as well as edition actions for the given
+resource (which usually includes calls to `PATCH/PUT resource/:id` endpoints).
+* user with `Owner` role can perform all the actions a `Contributor` can, as well as deleting actions for the given
+resource (which usually includes calls to `DELETE resource/:id` endpoints).
```

Comment:
What about `create` action? (POST project , POST scenario) 

What about `run marxan` action?

Reply:
This is a "conventions list", something I believe we will follow in most cases (I'll make this clearer). The detailed permissions matrix (a few lines below) is the place to go to check the comprehensive list of actions for each resource.

I didn't include `create` actions here, as the permissions for them might not be straightforward (i.e. we might want to allow anyone to create projects, or maybe just members of certain organizations, etc etc).

In any case, I'll add the actions you mentioned to the permission matrix :+1: